decidim-proposals/app/cells/decidim/proposals/proposal_m_cell.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+require "cell/partial"
+
+module Decidim
+  module Proposals
+    # This cell renders a proposal with its M-size card.
+    class ProposalMCell < Decidim::ViewModel
+      self.view_paths << "#{Decidim::Core::Engine.root}/app/views"
+      self.view_paths << "#{Decidim::Proposals::Engine.root}/app/cells"
+      self.view_paths << "#{Decidim::Proposals::Engine.root}/app/views"
+
+      include Decidim::LayoutHelper
+      include Decidim::ApplicationHelper
+      include Decidim::ActionAuthorizationHelper
+      include Decidim::TranslationsHelper
+      include Decidim::ResourceHelper
+      include Decidim::Proposals::ApplicationHelper
+      include Decidim::ResourceReferenceHelper
+
+      include Decidim::Proposals::Engine.routes.url_helpers
+
+      include Partial
+
+      def model_path
+        resource_locator(model).path
+      end
+
+      def current_settings
+        controller.current_settings
+      end
+
+      def feature_settings
+        controller.feature_settings
```

Comment:
I have a question about calls similar to this. If this cell lives in `decidim-proposals` component, and I want to render it from outside it, e.g. a `view_hook` from `decidim-participatory_processes`, this will work?

This will not return the settings of the component/feature that is being visited instead the settings of the feature that proposals belongs to?

Reply:
@rbngzlv you're right, this is gonna use whatever `feature_Settings` returns from the current controller rendering the cell.

What if we pass an `options` hash when rendering the view so we can overwrite this?

```ruby
def feature_settings
  options[:feature_setting] || controller.feature_settings
end
```

Reply:
so from the _outside_ of `decidim-proposals` the cell call could be something like:

```
<%= cell("decidim/proposals/proposal", model, context: {controller: controller}) %>
```

as the controller is needed for: `current_settings`, `feature_settings` and `action_authorization`

refactoring what @mrcasals stated before:

```
def controller
  options[:controller] || controller
end

def feature_settings
  controller.feature_settings
end
```

do you agree @mrcasals ?

Reply:
We can access `feature_settings` from the model? Something like:

```rb
def feature_settings
  model.feature.settings
end
```

I understood that the purpose of cells was to be self-sufficient and not depend on the context at all.

Reply:
We can re-think it: settings are used to decide whether a button should appear, or some conditions are checked or not. The cells should not care about those settings, but should know if a button should be rendered or not. Options that come to mind: 

1. pass a hash with options, using some sane defaults. Cells render different things according to the values of this hash
1. We have some "external" cell, and some "internal" cell. The external cell is how the cell should be rendered in other components, for example the search. When rendering a budget  project, we don't want to add it to a non-existing cart from the search, for example, as the cart logic belongs exclusively to the `budgets` component. Internal cell would be the one `budgets` use to render a project inside the component, so it knows about the cart.
1. A Combination of both: the internal cell receives a hash, the external cell renders the internal one setting the options automatically so we don't need to mess around from outside the component

@decidim/lot-core can you chime in?

Reply:
I agree with your proposal @mrcasals, specially option 3. I don't think is a good idea to allow carts to have all the actions in all the contexts (voting a proposal from the search results makes no sense to me), specially in the cases of `budgets` where there are technical issues that don't allow that.

Reply:
@mrcasals @oriolgual If you agree, in this first iteration, cards shown _outside_ its context will link to the card (XL) details.

Reply:
@agustibr the XL card is the one we use in the `show` views? If so, then I'd make them default to the M size instead, as it's much more reusable and we're already working on moving the M cards to cells, AFAIK

Reply:
Yes @mrcasals I think the default should be _M_ size, but i was talking about the link that appears in the footer of the M card

Reply:
Ah, I understand now, yeah go for it!